src/lib.rs

```diff
@@ -144,6 +145,24 @@ macro_rules! schema {
             fn incomplete_write(&self) -> bool {
                 self.incomplete_write
             }
+
```

Comment:
I don't think this guy should take a path, I don't think users need to decide where this location is. But also I don't think `writer` needs to take a path. Writer knows where it's file is, you can access that info.